jcache/embedded/src/main/java/org/infinispan/jcache/embedded/JCacheListenerAdapter.java

```diff
@@ -37,14 +37,14 @@ public void handleCacheEntryCreatedEvent(CacheEntryCreatedEvent<K, V> e) {
    public void handleCacheEntryModifiedEvent(CacheEntryModifiedEvent<K, V> e) {
       // JCache listeners notified only once, so do it after the event
       if (!e.isPre() && !e.isCreated())
-         notifier.notifyEntryUpdated(jcache, e.getKey(), e.getValue());
+         notifier.notifyEntryUpdated(jcache, e.getKey(), e.getValue(), e.getValue());
```

Comment:
Using current value with `isPre == false` is not correct, could you log a JIRA and add a reference here? I wonder why TCK did miss this.

Reply:
> I wonder why TCK did miss this.

Because the TCK only checks for a non-null value ;)

> Using current value with isPre == false is not correct.

Indeed, sending the current value as previous is not right. With the current embedded event API, getting the previous value would require handling isPre=true and somehow caching the value which would be messy. So, I guess you're referring to having a JIRA that adds a method to the event to retrieve the previous value independent of whether isPre or not?

Reply:
> Because the TCK only checks for a non-null value ;)
I guess we should file a PR to TCK...

> adds a method to the event to retrieve the previous value
Let's file a JIRA for flaw in our JCache impl ATM. You are describing a solution (probably the only viable). Caching the pre event is a no-go IMO.

Reply:
> I guess we should file a PR to TCK...

I've submitted [an issue](https://github.com/jsr107/jsr107tck/issues/139) for now.

> Caching the pre event is a no-go IMO.

+1

Reply:
> Let's file a JIRA for flaw in our JCache impl ATM

[ISPN-8652](https://issues.jboss.org/browse/ISPN-8652)